build/parseSpec.c

```diff
@@ -1128,10 +1103,74 @@ static rpmSpec parseSpec(const char *specFile, rpmSpecFlags flags,
     return NULL;
 }
 
+static rpmRC finalizeSpec(rpmSpec spec)
+{
+    rpmRC rc = RPMRC_FAIL;
+
+    /* XXX Skip valid arch check if not building binary package */
+    if (!(spec->flags & RPMSPEC_ANYARCH) && checkForValidArchitectures(spec)) {
+	goto exit;
+    }
+
+    fillOutMainPackage(spec->packages->header);
+    /* Define group tag to something when group is undefined in main package*/
+    if (!headerIsEntry(spec->packages->header, RPMTAG_GROUP)) {
+	headerPutString(spec->packages->header, RPMTAG_GROUP, "Unspecified");
+    }
+
+    char *platform = rpmExpand("%{_target_platform}", NULL);
+    char *os = rpmExpand("%{_target_os}", NULL);
+    char *optflags = rpmExpand("%{optflags}", NULL);
```

Comment:
These are still causing the warnings I mentioned earlier:

The uninitialized free warnings are still there:

```
/home/pmatilai/repos/rpm/build/parseSpec.c: In function ‘finalizeSpec’:
/home/pmatilai/repos/rpm/build/parseSpec.c:1160:5: warning: ‘platform’ may be used uninitialized [-Wmaybe-uninitialized]
 1160 |     free(platform);
      |     ^~~~~~~~~~~~~~
/home/pmatilai/repos/rpm/build/parseSpec.c:1121:11: note: ‘platform’ was declared here
 1121 |     char *platform = rpmExpand("%{_target_platform}", NULL);
      |           ^~~~~~~~
/home/pmatilai/repos/rpm/build/parseSpec.c:1161:5: warning: ‘os’ may be used uninitialized [-Wmaybe-uninitialized]
 1161 |     free(os);
      |     ^~~~~~~~
/home/pmatilai/repos/rpm/build/parseSpec.c:1122:11: note: ‘os’ was declared here
 1122 |     char *os = rpmExpand("%{_target_os}", NULL);
      |           ^~
/home/pmatilai/repos/rpm/build/parseSpec.c:1162:5: warning: ‘optflags’ may be used uninitialized [-Wmaybe-uninitialized]
 1162 |     free(optflags);
      |     ^~~~~~~~~~~~~~
/home/pmatilai/repos/rpm/build/parseSpec.c:1123:11: note: ‘optflags’ was declared here
 1123 |     char *optflags = rpmExpand("%{optflags}", NULL);
      |           ^~~~~~~~
```

Reply:
The reason these aren't showing up in CI is that CI is built without any optimizations.
Need to tweak that a bit...

Reply:
Okay, rebase to get the warnings -> compile fail on CI.